src/thread_pool.rs

```diff
@@ -19,10 +18,10 @@ pub struct Registry {
     thread_infos: Vec<ThreadInfo>,
     state: Mutex<RegistryState>,
     work_available: Condvar,
-    terminate: AtomicBool,
 }
 
 struct RegistryState {
+    terminate: bool,
```

Comment:
Nice, I like adding this here. Ensures we will see it eventually, but it doesn't interfere with the hot paths.